packages/app/src/cli/models/extensions/specification.ts

```diff
@@ -64,6 +64,8 @@ export interface ExtensionSpecification<TConfiguration extends BaseConfigType =
   transform?: (content: object) => object
   reverseTransform?: (content: object, options?: {flags?: Flag[]}) => object
   simplify?: (remoteConfig: SpecsAppConfiguration) => SpecsAppConfiguration
+  extensionManagedInToml?: boolean
+  multipleModuleConfigPath?: string
```

Comment:
since we want webhooks to have multiple modules, it's marked as an `extension` type in its specification. But we don't want the DX on app toml to change, so the first one is to indicate if there is extension data in the app toml. The `loader` loads all the extensions, but then it gets filtered out downstream based on `extension` type or `configuration` type - webhooklike when we are [creating `ExtensionInstances`](https://github.com/Shopify/cli/compare/main...extensions-configured-in-toml#diff-973c578c6a6700bec941fe428e1f488a4252f9fe356e5d002c6ce9d234137302R459). The flag lets us treat it as a `configuration` type on the CLI side since that's linked to app toml logic.

Maybe I should figure out a better name for `multipleModuleConfigPath` 😅 .. in the next PR, we want to get the config value from `webhook_subscription` specification so we can produce a module per subscription so `multipleModuleConfigPath` helps us fetch that by indicating the path: prototype is [here](https://github.com/Shopify/cli/compare/main...extensions-configured-in-toml#diff-ab169fa3e666a282671308e6c2af8e814783ccbdbfd7fe343128700590833c89R299)

Reply:
I still think we shouldn't need `extensionManagedInToml`, the specification shouldn't need to know where the data is stored. As an exercise, if we assume that the specification will be migrated to the API, would a field like this make sense in the API, or is it too specific for this use case?

The differenciation between configuration/extension might go away, and we can start breaking it in CLI: the app toml should be able to have non-config modules as well without the specification telling us so :) 

About `multipleModuleConfigPath` it's true is a complex use case 🤔 I would make the same thinking exercise, if we moved this field to the API, would it make sense or is it very specific for webhooks and CLI? Could it be used for other extensions as it is? I think we might need to rethink this before adding it to the spec :/ 

I can help with all this if needed! I just want to keep it clean instead of adding new properties to the specs.